### ISSUE TITLE
[Bug] failed to connect

### ISSUE THREAD

**Issue by GameOverFlowChart** [mlc-ai/mlc-llm#369]
The Android app was working for me. I downloaded the actual apk and replaced the older version on my phone with it (I deleted app data to be sure). In contrast to the old version this one doesn't automatically download the llm. If I try to add a model I get the error "add model failed, unable to connect to hugingface.co/18.154..."

The url Im trying is https://huggingface.co/mlc-ai/mlc-chat-vicuna-v1-7b-q3f16_0 because I guess that's the one the old version used.

Because it says that it's unable to connect I assume that it's a bug. I have internet access otherwise I couldn't write this bug report.

**Comment by cyx-6** [comment]
Thanks for reporting! Does this happen when downloading a builtin model, or manually adding the model by url?
And could you please provide the url you typed in?

**Comment by GameOverFlowChart** [comment]
I don't think that there is a build in model, at the top there is the word "model list" but nothing is listed. The Url I used is in my first post:
https://huggingface.co/mlc-ai/mlc-chat-vicuna-v1-7b-q3f16_0

I might be doing something wrong since this step was automatic in the older version of the app.

![Screenshot_2023-06-10-01-07-37-80_39ee8d39e9d9fa2b63462cd99e6b41fe](https://github.com/mlc-ai/mlc-llm/assets/109205200/265de498-fe49-4bd5-b948-12678531b6fb)

![Screenshot_2023-06-10-01-07-06-41_39ee8d39e9d9fa2b63462cd99e6b41fe](https://github.com/mlc-ai/mlc-llm/assets/109205200/95108c9f-2e1e-45de-8b29-366bd6ecd4a2)

**Comment by cyx-6** [comment]
oh, I think it will work if you change the url to
https://huggingface.co/mlc-ai/mlc-chat-vicuna-v1-7b-q3f16_0/blob/main/
which is the prefix of the
https://huggingface.co/mlc-ai/mlc-chat-vicuna-v1-7b-q3f16_0/blob/main/mlc-chat-config.json
we will add some information in the app for this.

**Comment by GameOverFlowChart** [comment]
I tried it with your version and I still have the same error.

![Screenshot_2023-06-10-01-15-37-84_39ee8d39e9d9fa2b63462cd99e6b41fe](https://github.com/mlc-ai/mlc-llm/assets/109205200/a85193a5-a3e4-49b3-b301-7be662141957)

![Screenshot_2023-06-10-01-15-53-62_39ee8d39e9d9fa2b63462cd99e6b41fe](https://github.com/mlc-ai/mlc-llm/assets/109205200/9f29da80-38d1-4750-8f6a-1c8cd0c069d6)

**Comment by cyx-6** [comment]
oh sorry, it should be
https://huggingface.co/mlc-ai/mlc-chat-vicuna-v1-7b-q3f16_0/resolve/main/
I mistakenly paste a wrong one.

**Comment by GameOverFlowChart** [comment]
That one also doesn't work I had found it in the source of the iOS app and tried it. Just to be sure I copy pasted it from your comment but same error as before.

![Screenshot_2023-06-10-01-23-55-37_39ee8d39e9d9fa2b63462cd99e6b41fe](https://github.com/mlc-ai/mlc-llm/assets/109205200/bb64ba35-5045-4bfa-b9d9-6d76db700e50)

**Comment by cyx-6** [comment]
By the way, why is the model list empty in your app, since the model deletion is still in progress.
Have you ever checked the networking permission is allowed for this app?

**Comment by GameOverFlowChart** [comment]
Yeah network is allowed.  I mean the app doesn't come with a model inside the apk does it? The apk is just 33MB. Yes the old version of the app had downloaded a model automatically but 1. The new version didn't list that model and 2. I cleared app data at some point  so even if the old model was there it should be gone, it's like a fresh install of the app.

**Comment by cyx-6** [comment]
It is wierd though, we do provide a builtin config
https://github.com/mlc-ai/mlc-llm/blob/main/android/MLCChat/app/src/main/assets/app-config.json
so there should be 2 models in model list, but not starting download files at all, execpt the `model-config.json`.
It may be possible that we failed to download those 2 `model-config.json` for 2 builtin models as well.
Can you visit the huggingface on your phone, without any VPN?

**Comment by GameOverFlowChart** [comment]
https://huggingface.co/mlc-ai/mlc-chat-vicuna-v1-7b-q3f16_0
![Screenshot_2023-06-10-01-41-54-34_438047d178ae0bdcccdfb4df8486f35b](https://github.com/mlc-ai/mlc-llm/assets/109205200/1196870a-707d-43e7-a23b-7d7b7b7e6974)

https://huggingface.co/mlc-ai/mlc-chat-vicuna-v1-7b-q3f16_0/resolve/main/
![Screenshot_2023-06-10-01-42-53-28_438047d178ae0bdcccdfb4df8486f35b](https://github.com/mlc-ai/mlc-llm/assets/109205200/38aa67ca-dad9-4c50-8735-521484e43967)

**Comment by GameOverFlowChart** [comment]
Also isn't it weird that there is an ip address inside the error message or is that normal?

**Comment by cyx-6** [comment]
the ip address is generated by system, and we directly print it out
https://github.com/mlc-ai/mlc-llm/blob/main/android/MLCChat/app/src/main/java/ai/mlc/mlcchat/AppViewModel.kt#L150
it might fail at this part
https://github.com/mlc-ai/mlc-llm/blob/main/android/MLCChat/app/src/main/java/ai/mlc/mlcchat/AppViewModel.kt#L114-L120
so that is expected that your model list is empty, becaust it failed to download all `model-config.json` from huggingface.
what is your phone model and system version? It would be great if you provide a screenshot in the system setting menu.
we will collect the information and try to reproduce if possible. And then get back to you if we figure out what happened here.

**Comment by GameOverFlowChart** [comment]
Problem solved. Since you said that there should be 2 models, I uninstalled and installed the app again. Now I see both models, and to test it I used https://huggingface.co/mlc-ai/mlc-chat-vicuna-v1-7b-q3f16_0/resolve/main/ and that got also added. Seems like the problem was that I updated from the older version to the newer, even deleting app data couldn't help in this specific case.

I guess the q4 model is the more powerful one, right?
